internal/api/option/log.go

```diff
@@ -81,18 +82,19 @@ func (i *logInterceptor) WrapStreamingHandler(
 		}
 
 		err := next(ctx, conn)
-		fields := log.Fields{
+		fields := map[string]any{
 			"connect.duration": time.Since(start),
 		}
-		level := log.InfoLevel
+		level := slog.LevelInfo
 		if err != nil {
-			level = log.ErrorLevel
+			level = slog.LevelInfo
 			fields["connect.code"] = connect.CodeOf(err).String()
 		}
 		logging.
 			LoggerFromContext(ctx).
-			WithFields(fields).
-			Log(level, "finished streaming call")
+			WithValues(fields).
+			V(int(level)).
+			Info("finished streaming call")
```

Comment:
I'm not sure this accomplishes the same thing as the old call. This change seems to use a logger with a particular verbosity, but doesn't seem to change the level of the message itself.

Reply:
Sorry, this one slipped me. I think the level is right, as I attempted to explain [here](https://github.com/akuity/kargo/pull/1791#discussion_r1569044829), but AFAIK there's no way to specify an error level log with `V()`.

I think the changes in [aa1a00d](https://github.com/akuity/kargo/pull/1791/commits/aa1a00dcc1d63480271ae7b336a786119784f938) should put us back on track.